src/history/history.service.ts

```diff
@@ -150,23 +151,19 @@ export class HistoryService {
     await this.connection.transaction(async (manager) => {
       const currentHistory = await manager.find<HistoryEntry>(HistoryEntry, {
         where: { user: user },
-        relations: ['note', 'user'],
+        relations: ['note', 'note.aliases', 'user'],
       });
       for (const entry of currentHistory) {
         await manager.remove<HistoryEntry>(entry);
       }
       for (const historyEntry of history) {
         this.notesService.checkNoteIdOrAlias(historyEntry.note);
-        const note = await manager.findOne<Note>(Note, {
-          where: [
-            {
-              id: historyEntry.note,
-            },
-            {
-              alias: historyEntry.note,
-            },
-          ],
-        });
+        const note = await manager
```

Comment:
We should probably call `getNoteByIdOrAlias` instead of replicating the query here

Reply:
We could do that, but I'm not sure if that is too wise as `getNoteByIdOrAlias` does 8 joins and a subquery which basically is what we do here. I'm uncertain what the best way forward is here…

Reply:
Let's leave it for now and wait for https://github.com/hedgedoc/hedgedoc/issues/1642, which should solve the issue